sdk/core/azure-core/CHANGELOG.md

```diff
@@ -22,6 +22,10 @@
 - Renamed `TelemetryPolicyOptions` to `TelemetryOptions`.
 - Renamed `ValuePolicyOptions` to `ValueOptions`.
 - Removed `StartTry()` from `Azure::Core::Http::Request`.
+- Changed type of `Azure::Core::Http::RetryOptions::StatusCodes` from `std::vector` to `std::set`.
```

Comment:
That's because repetition doesn't do anything, and order does not matter. Also, lookup is faster.
This is a tiny change, and I touched related code/neighboring lines.